src/parse/parse.c

```diff
@@ -6,64 +6,147 @@
-	}
-	head = NULL;
-	return (SUCCESS);
+    int fd;
+
+    if (current_token->type == RED_OUT)
+    {
+        fd = open(next_token->value, O_WRONLY | O_CREAT | O_TRUNC, 0664);
+        if (fd < 0)
+        {
+            perror("open");
+            return FAILURE;
+        }
+        enter_file(fd);
+        printf("%d\n",fd);
+        //printf("\n");
+        return SUCCESS;
+    }
+    else if (current_token->type == APPEND)
+    {
+        fd = open(next_token->value, O_WRONLY | O_CREAT | O_APPEND, 0664);
+        if (fd < 0)
+        {
+            perror("open");
+            return FAILURE;
+        }
+        enter_file(fd);
+        printf("%d\n",fd);
+        //printf("\n");
+        return SUCCESS;
+    }
+    return FAILURE;
 }
-/*
-int parse(t_data *data)
-{
-    int status;
-    status = 0;
 
-    //status = tokens_parsing(data->token);
-    return (status);
+
+int handle_input_redirection(t_token *token, t_token *next_token)
+{
+    if (ft_strcmp(token->value, "<") == 0)
+    {
+        open(next_token->value, O_RDONLY);
+        printf("\n"); // idk if we actually need to print this new line
+        return (0);
+    }
+    return FAILURE;
 }
-*/
 
-		// if ((head->type == REDIRECT_IN && next_type == COMMAND)
-		// 	|| (head->type == REDIRECT_IN && next_type == REDIRECT_IN)
-		// 	|| (head->type == REDIRECT_IN && next_type == REDIRECT_OUT)
-		// 	|| (head->type == REDIRECT_IN && next_type == PIPE)
-		// 	|| (head->type == REDIRECT_IN && next_type == HEREDOC)
-		// 	|| (head->type == REDIRECT_IN && next_type == APPEND)
-		// 	|| (head->type == REDIRECT_IN && next_type == FLAG))
\ No newline at end of file
+
+
+int parse_token(t_token *token)
+{
+    t_token *current_token = token;
+
+    while (current_token != NULL)
+    {
+        // Handle output redirection (">" or ">>")
+        if ((current_token->type == RED_OUT || current_token->type == APPEND)
+            && current_token->next != NULL && current_token->next->value != NULL)
+        {
+            if (current_token->prev == NULL) // No previous token
+            {
+                if (current_token->next->type == ARGUMENT)
+                    handle_output_redirection(current_token, current_token->next);
+                else
+                {
+                    printf("error 106\n");
+                    return 44;
+                }
+            }
+            else // Previous token exists
+            {
+                if (current_token->next->type == ARGUMENT && current_token->prev->type == PIPE)
+                    handle_output_redirection(current_token, current_token->next);
+                else
+                {
+                    printf("error 117\n");
+                    return 44;
+                }
+            }
+        }
+        // Handle input redirection ("<" or "<<")
+        else if ((current_token->type == RED_IN || current_token->type == HEREDOC)
+            && current_token->next != NULL && current_token->next->value != NULL)
```

Comment:
If the next node is not NULL, very much possibly the value (current_token->next->value) will not be null. Even if it is, we should probably take it as an argument, because null arguments are still a thing.